controllers/nificluster_controller.go

```diff
@@ -181,11 +175,13 @@ func (r *NifiClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
-	if err := k8sutil.UpdateCRStatus(r.Client, instance, v1alpha1.NifiClusterRunning, r.Log); err != nil {
-		return RequeueWithError(r.Log, err.Error(), err)
+	if !instance.IsReady() {
+		if err := k8sutil.UpdateCRStatus(r.Client, instance, v1alpha1.NifiClusterRunning, r.Log); err != nil {
+			return RequeueWithError(r.Log, err.Error(), err)
+		}
 	}
 
-	return Reconciled()
+	return RequeueAfter(intervalRunning)
```

Comment:
@cannonpalms asked:

> Can you speak to the strategy for watching child resources (namely pods) now that the NifiCluster controller is not flapping?

Prior to this change, the `NifiCluster` controller perpetually scheduled reconciliation requests by updating its own status (`Reconciling` vs `Running`). The controller still watches all of the resources it was watching before. The difference is that now the controller schedules reconciliation requests respecting its reconciliation interval of `15s` -- that's the change on this line of code. Reconciliation requests will also be triggered if the `NifiCluster` CRD is modified by a user.

The same is true for the `NifiDataflow` controller.